walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,156 @@
+   * @param request This is a meeting request that is to be scheduled
+   * @return a collection of all possible times that {@code request} can be scheduled
+   */
+  private Collection<TimeRange> queryHelper(Collection<Event> events, MeetingRequest request) {
+    // Define the variables to be used
+    int START_OF_DAY = TimeRange.START_OF_DAY;
+    int END_OF_DAY = TimeRange.END_OF_DAY;
+    long durationOfMeeting = request.getDuration();
+
+    // List of all possible time ranges that attendees can have a meeting
+    Collection<TimeRange> possibleTimes = new ArrayList<>();
+    // Create a list of attendees and add all the attendees to that list
+    Collection<String> attendees = new ArrayList<>();
+    attendees.addAll(request.getAttendees());
+
+    // Collection of timeranges that can not work for the proposed schedule
+    // These are times when the attendees have an event scheduled at that particular time
+    List<TimeRange> attendeesCannotScheduleHere = cannotSheduleMeeting(attendees, events); 
+
+    // Return the whole day if all attendees are free for the whole day
+    if (attendeesCannotScheduleHere.isEmpty()) 
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+
+    // Sort the time ranges by the end time in ascending order
+    Collections.sort(attendeesCannotScheduleHere, TimeRange.ORDER_BY_END);
+
+    // Remove the time ranges which are part of longer time ranges
+    removeNestedTimes(attendeesCannotScheduleHere);
+
+    for (int index = 0; index < attendeesCannotScheduleHere.size(); index++) {
+      TimeRange currentTimeRange = attendeesCannotScheduleHere.get(index);
+      int startOfCurrentTimeRange = currentTimeRange.start();
+      int endOfCurrentTimeRange = currentTimeRange.end();
+
+      // We have 3 different ways we can schedule the meeting at according to the list of time ranges
+      // 1. When we have the first end time, we can schedule before it's start time and start of day. It is also the first start time because we removed the nested times
+      // 2. Times which are in between the first and last end time in the time ranges
+      // 3. The last end time, we can check if we can schedule an item between it and the end of day
+      // First(1) way
+      if (index == 0) {
+        addPossibleTime(START_OF_DAY, startOfCurrentTimeRange, false, durationOfMeeting, possibleTimes);
+      }
+      // Second(2) way
+      if (index +1 < attendeesCannotScheduleHere.size()) {
+        TimeRange currentTimeRangePlusOne = attendeesCannotScheduleHere.get(index + 1);
+        addPossibleTime(endOfCurrentTimeRange, currentTimeRangePlusOne.start(), false, durationOfMeeting, possibleTimes);
+      }
+      //Third(3) way
+      if (index == attendeesCannotScheduleHere.size() - 1) {
+        addPossibleTime(endOfCurrentTimeRange, END_OF_DAY, true, durationOfMeeting, possibleTimes);
+      }
+    }
+    return possibleTimes;
+  }
+
+  /**
+   * Create and check if timerange is appropriate for scheduling requesting time
```

Comment:
nit: might be helpful to comment that if an available time is found, it will be added to the possibleTimes param